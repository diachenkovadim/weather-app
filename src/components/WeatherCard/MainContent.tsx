import { FC } from 'react';
import { Stack, Typography } from '@mui/material';

import { roundNumber } from 'utils/formatNumber';
import { fDate } from 'utils/formatDate';

import { IMainContentProps } from './weatherCard.types';
import { ImageBox } from '../ImageBox';

export const MainContent: FC<IMainContentProps> = (props) => {
  const {
    currentTemperature,
    icon,
    maxTemperature,
    minTemperature,
    weatherDescription,
    timeStamp,
  } = props;

  const date = fDate(timeStamp * 1000);
  return (
    <Stack
      direction={{ xs: 'column-reverse', sm: 'row' }}
      spacing={1}
      justifyContent="space-between"
    >
      <Stack spacing={1}>
        <Stack direction="row" spacing={1} justifyContent="space-between">
          <Typography variant="body2">Date:</Typography>
          <Typography variant="subtitle2">{date}</Typography>
        </Stack>
        <Stack direction="row" spacing={1} justifyContent="space-between">
          <Typography variant="body2">Current Temperature:</Typography>
          <Typography variant="subtitle2">
            {roundNumber(currentTemperature)} °C
          </Typography>
        </Stack>

        <Stack direction="row" spacing={1} justifyContent="space-between">
          <Typography variant="body2"> Min. Temperature:</Typography>
          <Typography variant="subtitle2">
            {roundNumber(minTemperature)} °C
          </Typography>
        </Stack>

        <Stack direction="row" spacing={1} justifyContent="space-between">
          <Typography variant="body2"> Max. Temperature:</Typography>
          <Typography variant="subtitle2">
            {roundNumber(maxTemperature)} °C
          </Typography>
        </Stack>
      </Stack>

      <Stack alignItems="center">
        <Typography
          variant="subtitle2"
          textTransform="capitalize"
          textAlign="center"
        >
          {weatherDescription}
        </Typography>
        <ImageBox imagePath={icon} width={100} height={100} />
      </Stack>
    </Stack>
  );
};
