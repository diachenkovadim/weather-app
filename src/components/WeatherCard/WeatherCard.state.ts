import { useState } from 'react';
import { useNavigate } from 'react-router-dom';

import {
  fetchWeatherData,
  removeWeatherCard,
  selectWeatherCard,
} from 'services/redux/weatherReducer';
import { useAppDispatch } from 'hooks/useRedux';

import { IuseWeatherCardStateProps } from './weatherCard.types';

import { WEATHER_DETAILS } from 'constants/routes';

export const useWeatherCardState = (props: IuseWeatherCardStateProps) => {
  const { lat, lon, cardId } = props;

  const navigate = useNavigate();
  const dispatch = useAppDispatch();
  const [isLoading, setIsLoading] = useState(false);

  const onDeleteWeatherCardHandler = (
    event: React.MouseEvent<HTMLSpanElement>
  ) => {
    event.stopPropagation();
    cardId && dispatch(removeWeatherCard(cardId));
  };

  const onRefreshWeatherCardDataHandler = async (
    event: React.MouseEvent<HTMLSpanElement>
  ) => {
    try {
      event.stopPropagation();
      setIsLoading(true);
      await dispatch(fetchWeatherData({ lat, lon }));
      setIsLoading(false);
    } catch (error) {
      setIsLoading(false);
    }
  };

  const onCardClickHandler = () => {
    dispatch(selectWeatherCard(cardId));
    navigate(`${WEATHER_DETAILS}${cardId}`);
  };

  return {
    onDeleteWeatherCardHandler,
    onRefreshWeatherCardDataHandler,
    onCardClickHandler,
    isLoading,
  };
};
