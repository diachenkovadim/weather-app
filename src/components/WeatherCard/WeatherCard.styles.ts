import { Box, alpha, styled } from '@mui/material';

export const WeatherCardStyles = {
  StyledBox: styled(Box)(({ theme }) => ({
    display: 'flex',
    ':hover': {
      boxShadow: theme.customShadows.primary,
    },
    padding: 5,
    borderRadius: 8,
    backgroundColor: alpha(theme.palette.common.white, 0.5),
  })),
};
