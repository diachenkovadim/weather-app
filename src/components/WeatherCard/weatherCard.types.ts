export interface IWeatherCardProps {
  cityName: string;
  currentTemperature: number;
  minTemperature: number;
  maxTemperature: number;
  icon: string;
  country: string;
  timeStamp: number;
  weatherDescription: string;
  cardId: number;
  lat: number;
  lon: number;
}

export interface IMainContentProps
  extends Omit<
    IWeatherCardProps,
    'cityName' | 'lat' | 'lon' | 'country' | 'cardId'
  > {}

export interface IuseWeatherCardStateProps
  extends Pick<IWeatherCardProps, 'lat' | 'lon' | 'cardId'> {}
