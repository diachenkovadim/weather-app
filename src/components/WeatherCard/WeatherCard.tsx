import { FC } from 'react';
import { CardContent, Stack, Typography } from '@mui/material';
import RefreshRoundedIcon from '@mui/icons-material/RefreshRounded';
import DeleteForeverRoundedIcon from '@mui/icons-material/DeleteForeverRounded';

import { StyledCard } from 'styles/overrides/Card';

import { MainContent } from './MainContent';
import { IWeatherCardProps } from './weatherCard.types';
import { useWeatherCardState } from './WeatherCard.state';
import { WeatherCardSkeleton } from '../Skeleton';
import { WeatherCardStyles as Styled } from './WeatherCard.styles';

export const WeatherCard: FC<IWeatherCardProps> = (props) => {
  const {
    cityName,
    currentTemperature,
    country,
    timeStamp,
    icon,
    maxTemperature,
    minTemperature,
    weatherDescription,
    cardId,
    lat,
    lon,
  } = props;

  const {
    isLoading,
    onCardClickHandler,
    onDeleteWeatherCardHandler,
    onRefreshWeatherCardDataHandler,
  } = useWeatherCardState({ lat, lon, cardId });
  return (
    <StyledCard
      sx={{
        background: (theme) =>
          `linear-gradient(to right, ${theme.palette.success.lighter}, ${theme.palette.success.light})`,
      }}
      onClick={onCardClickHandler}
      data-testid="weather-card"
    >
      <CardContent>
        {isLoading ? (
          <WeatherCardSkeleton />
        ) : (
          <>
            <Stack direction="row" justifyContent="space-between" mb={1}>
              <Typography variant="h5">
                {cityName}, {country}
              </Typography>
              <Stack direction="row" spacing={1} alignItems="center">
                <Styled.StyledBox
                  component="span"
                  data-testid="refresh-button"
                  onClick={onRefreshWeatherCardDataHandler}
                >
                  <RefreshRoundedIcon />
                </Styled.StyledBox>
                <Styled.StyledBox
                  data-testid="remove-button"
                  onClick={onDeleteWeatherCardHandler}
                >
                  <DeleteForeverRoundedIcon />
                </Styled.StyledBox>
              </Stack>
            </Stack>

            <MainContent
              currentTemperature={currentTemperature}
              timeStamp={timeStamp}
              icon={icon}
              maxTemperature={maxTemperature}
              minTemperature={minTemperature}
              weatherDescription={weatherDescription}
            />
          </>
        )}
      </CardContent>
    </StyledCard>
  );
};
