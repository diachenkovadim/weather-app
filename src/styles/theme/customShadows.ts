import { alpha } from '@mui/material/styles';

import { palette } from './palette';

interface CustomShadowOptions {
  primary: string;
  info: string;
  card: string;
}

declare module '@mui/material/styles' {
  interface Theme {
    customShadows: CustomShadowOptions;
  }
  interface ThemeOptions {
    customShadows?: CustomShadowOptions;
  }
}

const themeColor = palette();

const LIGHT_MODE = themeColor.grey[500];

const createShadow = (color: string) => {
  return {
    primary: `0 8px 16px 0 ${alpha(palette().primary.main, 0.24)}`,
    info: `0 8px 16px 0 ${alpha(palette().info.main, 0.24)}`,
    card: `0 0 2px 0 ${alpha(color, 0.2)}, 0 12px 24px -4px ${alpha(
      color,
      0.12
    )}`,
  };
};

export const customShadows = () => createShadow(LIGHT_MODE);
