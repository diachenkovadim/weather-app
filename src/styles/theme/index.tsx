import { useMemo } from 'react';

import { CssBaseline } from '@mui/material';
import {
  ThemeProvider as MUIThemeProvider,
  StyledEngineProvider,
  ThemeOptions,
  createTheme,
} from '@mui/material/styles';

import { customShadows } from './customShadows';
import { GlobalStyles } from './globalStyles';
import { palette } from './palette';
import { shadows } from './shadows';
import { typography } from './typography';

type TProps = {
  children: React.ReactNode;
};

export const ThemeProvider = ({ children }: TProps) => {
  const themeOptions: ThemeOptions = useMemo(
    () => ({
      palette: palette(),
      typography,
      shape: { borderRadius: 8 },
      shadows: shadows(),
      customShadows: customShadows(),
    }),
    []
  );

  const theme = createTheme(themeOptions);

  return (
    <StyledEngineProvider injectFirst>
      <MUIThemeProvider theme={theme}>
        <CssBaseline />
        <GlobalStyles />
        {children}
      </MUIThemeProvider>
    </StyledEngineProvider>
  );
};
