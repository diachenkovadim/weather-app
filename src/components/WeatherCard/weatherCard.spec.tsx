import { render, screen, fireEvent } from '@testing-library/react';

import { WeatherCard } from './WeatherCard';
import { useWeatherCardState } from './WeatherCard.state';
import { ThemeProvider } from '../../styles/theme';

jest.mock('./WeatherCard.state', () => ({
  useWeatherCardState: jest.fn(),
}));

const mockProps = {
  cityName: 'Kharkiv',
  currentTemperature: 20,
  country: 'UA',
  timeStamp: 1620399000,
  icon: '01d',
  maxTemperature: 22,
  minTemperature: 18,
  weatherDescription: 'clear sky',
  cardId: 1,
  lat: 48.856614,
  lon: 2.3522219,
};
const mockOnRefresh = jest.fn();
const mockOnDelete = jest.fn();
const mockOnCardClick = jest.fn();

const mockState = {
  onDeleteWeatherCardHandler: mockOnDelete,
  onRefreshWeatherCardDataHandler: mockOnRefresh,
  onCardClickHandler: mockOnCardClick,
  isLoading: false,
};
describe('WeatherCard', () => {
  beforeEach(() => {
    const mockUseWeatherCardState = jest.fn().mockReturnValue(mockState);

    (useWeatherCardState as jest.Mock).mockImplementation(
      mockUseWeatherCardState
    );
  });

  it('should render the weather card component', () => {
   const view = render(
      <ThemeProvider>
        <WeatherCard {...mockProps} />;
      </ThemeProvider>
    );
    expect(view).toMatchSnapshot()
    expect(screen.getByTestId('weather-card')).toBeInTheDocument();
  });

  it('should render WeatherCard with props', () => {
    render(
      <ThemeProvider>
        <WeatherCard {...mockProps} />;
      </ThemeProvider>
    );

    expect(
      screen.getByText(`${mockProps.cityName}, ${mockProps.country}`)
    ).toBeInTheDocument();
    expect(
      screen.getByText(`${mockProps.weatherDescription}`)
    ).toBeInTheDocument();
  });

  it('refresh button handler should works', () => {
    render(
      <ThemeProvider>
        <WeatherCard {...mockProps} />;
      </ThemeProvider>
    );

    const button = screen.getByTestId('refresh-button');

    expect(button).toBeInTheDocument();
    fireEvent.click(button);
    expect(mockOnRefresh).toHaveBeenCalledTimes(1);
  });

  it('click on card handler should works', () => {
    render(
      <ThemeProvider>
        <WeatherCard {...mockProps} />;
      </ThemeProvider>
    );

    const card = screen.getByTestId('weather-card');

    expect(card).toBeInTheDocument();
    fireEvent.click(card);
    expect(mockOnCardClick).toHaveBeenCalledTimes(1);
  });

  it('remove button handler should works', () => {
    render(
      <ThemeProvider>
        <WeatherCard {...mockProps} />;
      </ThemeProvider>
    );

    const button = screen.getByTestId('remove-button');

    expect(button).toBeInTheDocument();
    fireEvent.click(button);
    expect(mockOnDelete).toHaveBeenCalledTimes(1);
  });
});
