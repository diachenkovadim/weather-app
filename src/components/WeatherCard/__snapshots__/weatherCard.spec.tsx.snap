// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherCard should render the weather card component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-relic7-MuiPaper-root-MuiCard-root"
        data-testid="weather-card"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <div
            class="MuiStack-root css-k8oxap-MuiStack-root"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 css-s3e1oj-MuiTypography-root"
            >
              Kharkiv
              , 
              UA
            </h5>
            <div
              class="MuiStack-root css-1q36rmd-MuiStack-root"
            >
              <span
                class="MuiBox-root css-1xhjc5q"
                data-testid="refresh-button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="RefreshRoundedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M17.65 6.35c-1.63-1.63-3.94-2.57-6.48-2.31-3.67.37-6.69 3.35-7.1 7.02C3.52 15.91 7.27 20 12 20c3.19 0 5.93-1.87 7.21-4.56.32-.67-.16-1.44-.9-1.44-.37 0-.72.2-.88.53-1.13 2.43-3.84 3.97-6.8 3.31-2.22-.49-4.01-2.3-4.48-4.52C5.31 9.44 8.26 6 12 6c1.66 0 3.14.69 4.22 1.78l-1.51 1.51c-.63.63-.19 1.71.7 1.71H19c.55 0 1-.45 1-1V6.41c0-.89-1.08-1.34-1.71-.71l-.64.65z"
                  />
                </svg>
              </span>
              <div
                class="MuiBox-root css-1xhjc5q"
                data-testid="remove-button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="DeleteForeverRoundedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zm3.17-6.41a.9959.9959 0 0 1 0-1.41c.39-.39 1.02-.39 1.41 0L12 12.59l1.41-1.41c.39-.39 1.02-.39 1.41 0s.39 1.02 0 1.41L13.41 14l1.41 1.41c.39.39.39 1.02 0 1.41s-1.02.39-1.41 0L12 15.41l-1.41 1.41c-.39.39-1.02.39-1.41 0a.9959.9959 0 0 1 0-1.41L10.59 14l-1.42-1.41zM18 4h-2.5l-.71-.71c-.18-.18-.44-.29-.7-.29H9.91c-.26 0-.52.11-.7.29L8.5 4H6c-.55 0-1 .45-1 1s.45 1 1 1h12c.55 0 1-.45 1-1s-.45-1-1-1z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="MuiStack-root css-m9sgp4-MuiStack-root"
          >
            <div
              class="MuiStack-root css-1okj3ks-MuiStack-root"
            >
              <div
                class="MuiStack-root css-jibmk2-MuiStack-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-1o1ohr1-MuiTypography-root"
                >
                  Date:
                </p>
                <h6
                  class="MuiTypography-root MuiTypography-subtitle2 css-1scmp52-MuiTypography-root"
                >
                  07/05/21 17:50
                </h6>
              </div>
              <div
                class="MuiStack-root css-jibmk2-MuiStack-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-1o1ohr1-MuiTypography-root"
                >
                  Current Temperature:
                </p>
                <h6
                  class="MuiTypography-root MuiTypography-subtitle2 css-1scmp52-MuiTypography-root"
                >
                  20
                   °C
                </h6>
              </div>
              <div
                class="MuiStack-root css-jibmk2-MuiStack-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-1o1ohr1-MuiTypography-root"
                >
                   Min. Temperature:
                </p>
                <h6
                  class="MuiTypography-root MuiTypography-subtitle2 css-1scmp52-MuiTypography-root"
                >
                  18
                   °C
                </h6>
              </div>
              <div
                class="MuiStack-root css-jibmk2-MuiStack-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body2 css-1o1ohr1-MuiTypography-root"
                >
                   Max. Temperature:
                </p>
                <h6
                  class="MuiTypography-root MuiTypography-subtitle2 css-1scmp52-MuiTypography-root"
                >
                  22
                   °C
                </h6>
              </div>
            </div>
            <div
              class="MuiStack-root css-163ljxj-MuiStack-root"
            >
              <h6
                class="MuiTypography-root MuiTypography-subtitle2 css-1ks3vq0-MuiTypography-root"
              >
                clear sky
              </h6>
              <img
                alt="weather picture"
                class="MuiBox-root css-99hf7"
                src="01d.png"
              />
            </div>
          </div>
        </div>
      </div>
      ;
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-relic7-MuiPaper-root-MuiCard-root"
      data-testid="weather-card"
    >
      <div
        class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <div
          class="MuiStack-root css-k8oxap-MuiStack-root"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 css-s3e1oj-MuiTypography-root"
          >
            Kharkiv
            , 
            UA
          </h5>
          <div
            class="MuiStack-root css-1q36rmd-MuiStack-root"
          >
            <span
              class="MuiBox-root css-1xhjc5q"
              data-testid="refresh-button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="RefreshRoundedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M17.65 6.35c-1.63-1.63-3.94-2.57-6.48-2.31-3.67.37-6.69 3.35-7.1 7.02C3.52 15.91 7.27 20 12 20c3.19 0 5.93-1.87 7.21-4.56.32-.67-.16-1.44-.9-1.44-.37 0-.72.2-.88.53-1.13 2.43-3.84 3.97-6.8 3.31-2.22-.49-4.01-2.3-4.48-4.52C5.31 9.44 8.26 6 12 6c1.66 0 3.14.69 4.22 1.78l-1.51 1.51c-.63.63-.19 1.71.7 1.71H19c.55 0 1-.45 1-1V6.41c0-.89-1.08-1.34-1.71-.71l-.64.65z"
                />
              </svg>
            </span>
            <div
              class="MuiBox-root css-1xhjc5q"
              data-testid="remove-button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="DeleteForeverRoundedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zm3.17-6.41a.9959.9959 0 0 1 0-1.41c.39-.39 1.02-.39 1.41 0L12 12.59l1.41-1.41c.39-.39 1.02-.39 1.41 0s.39 1.02 0 1.41L13.41 14l1.41 1.41c.39.39.39 1.02 0 1.41s-1.02.39-1.41 0L12 15.41l-1.41 1.41c-.39.39-1.02.39-1.41 0a.9959.9959 0 0 1 0-1.41L10.59 14l-1.42-1.41zM18 4h-2.5l-.71-.71c-.18-.18-.44-.29-.7-.29H9.91c-.26 0-.52.11-.7.29L8.5 4H6c-.55 0-1 .45-1 1s.45 1 1 1h12c.55 0 1-.45 1-1s-.45-1-1-1z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          class="MuiStack-root css-m9sgp4-MuiStack-root"
        >
          <div
            class="MuiStack-root css-1okj3ks-MuiStack-root"
          >
            <div
              class="MuiStack-root css-jibmk2-MuiStack-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-1o1ohr1-MuiTypography-root"
              >
                Date:
              </p>
              <h6
                class="MuiTypography-root MuiTypography-subtitle2 css-1scmp52-MuiTypography-root"
              >
                07/05/21 17:50
              </h6>
            </div>
            <div
              class="MuiStack-root css-jibmk2-MuiStack-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-1o1ohr1-MuiTypography-root"
              >
                Current Temperature:
              </p>
              <h6
                class="MuiTypography-root MuiTypography-subtitle2 css-1scmp52-MuiTypography-root"
              >
                20
                 °C
              </h6>
            </div>
            <div
              class="MuiStack-root css-jibmk2-MuiStack-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-1o1ohr1-MuiTypography-root"
              >
                 Min. Temperature:
              </p>
              <h6
                class="MuiTypography-root MuiTypography-subtitle2 css-1scmp52-MuiTypography-root"
              >
                18
                 °C
              </h6>
            </div>
            <div
              class="MuiStack-root css-jibmk2-MuiStack-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-1o1ohr1-MuiTypography-root"
              >
                 Max. Temperature:
              </p>
              <h6
                class="MuiTypography-root MuiTypography-subtitle2 css-1scmp52-MuiTypography-root"
              >
                22
                 °C
              </h6>
            </div>
          </div>
          <div
            class="MuiStack-root css-163ljxj-MuiStack-root"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle2 css-1ks3vq0-MuiTypography-root"
            >
              clear sky
            </h6>
            <img
              alt="weather picture"
              class="MuiBox-root css-99hf7"
              src="01d.png"
            />
          </div>
        </div>
      </div>
    </div>
    ;
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
